test/testdata/infer/generics/bounds.rb

```diff
@@ -0,0 +1,35 @@
+# typed: true
+
+class Animal; end
+class Cat < Animal; end
+class Persian < Cat; end
+
+class A1
+  extend T::Sig
+  extend T::Generic
+
+  X = type_member(fixed: Cat, lower: Persian, upper: Animal)
+    # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ error: Type member is defined with bounds and `:fixed`
+end
+
+class A2
+  extend T::Sig
+  extend T::Generic
+
+  # This should allow instantiations of this variable as any of Animal, Cat, or
+  # Persian.
+  X = type_member(lower: Persian, upper: Animal)
```

Comment:
could we make this into an error before we implement it properly?

Reply:
Good idea. I've added the error and updated the tests to check for that.